packages/kbn-test/src/page_load_metrics/navigation.ts

```
@@ -0,0 +1,159 @@
+
+  await Promise.all(
+    apps.map(async (app) => {
+      const page = await browser.newPage();
+      page.setCacheEnabled(false);
+      page.setDefaultNavigationTimeout(0);
+      const frameResponses = new Map<string, FrameResponse>();
+      devToolsResponses.set(app.path, frameResponses);
+
+      const client = await page.target().createCDPSession();
+      await client.send('Network.enable');
+
+      function getRequestData(requestId: string) {
+        if (!frameResponses.has(requestId)) {
+          frameResponses.set(requestId, { url: '', dataLength: 0 });
+        }
+
+        return frameResponses.get(requestId)!;
+      }
+
+      client.on('Network.responseReceived', (event: ResponseReceivedEvent) => {
+        getRequestData(event.requestId).url = event.response.url;
+      });
+
+      client.on('Network.dataReceived', (event: DataReceivedEvent) => {
+        getRequestData(event.requestId).dataLength += event.dataLength;
+      });
+
+      const url = createUrl(app.path, options.appConfig.url);
+      log.debug(`goto ${url}`);
+      await page.goto(url, {
+        waitUntil: 'networkidle0',
+      });
+
+      let readyAttempt = 0;
+      let selectorFound = false;
+      while (!selectorFound) {
+        readyAttempt += 1;
+        try {
+          await page.waitForSelector(app.locator, { timeout: 5000 });
+          selectorFound = true;
+        } catch (error) {
+          log.error(
+            `Page '${app.path}' was not loaded properly, unable to find '${
+              app.locator
+            }', url: ${page.url()}`
+          );
+
+          if (readyAttempt < 6) {
+            continue;
+          }
+
+          await page.bringToFront();
+          Fs.mkdirSync(failureDir, { recursive: true });
+          await page.screenshot({
+            path: resolve(failureDir, `${app.path.replace('/', '_')}_navigation.png`),
+            type: 'png',
+            fullPage: true,
+          });
+
+          throw error;
```

I think we should throw a different error with a better error message, the default message is a little misleading as it seems the timeout is only 5 seconds